File: wiz-baseball-game-java/src/main/java/BaseballGameWithZero.java
import java.util.List;

public class BaseballGameWithZero extends BaseballGame {

	public BaseballGameWithZero() {
		super(List.of('1', '2', '3', '4', '5', '6', '7', '8', '9', '0'));
	}

	public BaseballGameWithZero(int size) {
		super(size, List.of('1', '2', '3', '4', '5', '6', '7', '8', '9', '0'));
	}

	@Override
	public String getResult() {
		return "[0포함 모드] " + super.getResult();
	}
}

File: wiz-baseball-game/src/main/kotlin/BaseballGameWithZero.kt
import kotlin.random.Random

class BaseballGameWithZero : BaseballGame() {

    override val answer: List<Int>

    init {
        val first = (1..9).random()
        answer = listOf(first) +
                (0..9).filterNot { it == first }
                    .shuffled()
                    .take(size - 1)
    }

    override fun checkInput(input: List<Int>): Boolean {
        if (input[0] == 0) return false
        return super.checkInput(input)
    }
}